checks/frozen_deps.go

```diff
@@ -188,40 +213,80 @@ func validateDockerfile(path string, content []byte,
 	return ret, nil
 }
 
-// ============================================================
-// ===================== Github workflows =====================
-// ============================================================.
+func isGitHubWorkflowScriptFreeOfInsecureDownloads(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(CheckFrozenDeps, "*", false, c, validateGitHubWorkflowShellScriptDownloads)
+}
+
+func validateGitHubWorkflowShellScriptDownloads(pathfn string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	if len(content) == 0 {
+		return false, ErrEmptyFile
+	}
+
+	var workflow gitHubActionWorkflowConfig
+	err := yaml.Unmarshal(content, &workflow)
+	if err != nil {
+		return false, fmt.Errorf("!! frozen-deps - cannot unmarshal file %v\n%v: %w", pathfn, string(content), err)
+	}
+
+	githubVarRegex := regexp.MustCompile(`{{[^{}]*}}`)
+	validated := true
+	// https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell.
```

Comment:
do we need to check the workflow's platform to make sure it's not windows?

otherwise the windows default shell is `pwsh` but we continue trying to analyze it under current logic (and potentially add unhelpful error messages)?

Reply:
good point. `pwsh` is not supported as a shell, see https://github.com/ossf/scorecard/pull/610/files#diff-5345e5b2866b919ef20545abf37043d1ee8ba7a30b8e8d27a1f7a9ded375232aR50
I'll add a comment